Sources/UtiliKit/ActiveLabel/ActiveLabel.swift

```diff
@@ -0,0 +1,324 @@
+            view.backgroundColor = loadingViewColor
+            
+            let maskLayer = createLayerMask()
+            maskLayer.add(maskAnimation(), forKey: "maskAnimation")
+            view.layer.mask = maskLayer
+            loadingViews.append(view)
+            
+            addSubview(view)
+            setupConstraints(for: view, at: index)
+        }
+        
+        loadingViews.forEach({ $0.isHidden = isHidden })
+    }
+    
+    func setupConstraints(for view: UIView, at index: Int) {
+        NSLayoutConstraint.activate([view.leadingAnchor.constraint(equalTo: leadingAnchor),
+                                     view.heightAnchor.constraint(greaterThanOrEqualToConstant: loadingLineHeight)])
+        
+        if index == 0 {
+            // Constrain the first view to the top of the label.
+            view.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        } else {
+            // If not the first view constrain the top to the above views bottom with the loadingLineVerticalSpacing.
+            let aboveView = loadingViews[index - 1]
+            view.topAnchor.constraint(equalTo: aboveView.bottomAnchor, constant: loadingLineVerticalSpacing).isActive = true
+        }
+        
+        if index + 1 == estimatedNumberOfLines {
+            // If this is the last view then constraint it's bottom to the label
+            view.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+            
+            // Handle the last line width/trailing constraint based on finalLineTrailingInset, finalLineLength, or none in that order.
+            if finalLineTrailingInset != 0 {
+                view.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -finalLineTrailingInset).isActive = true
+            } else if finalLineLength != 0 {
+                view.widthAnchor.constraint(equalToConstant: finalLineLength).isActive = true
+            } else {
+                view.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
+            }
+        } else {
+            // Only set up the trailing constraint since next view will constrain to this bottomAnchor.
+            view.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
+        }
+    }
+    
+    // MARK: - CA Helpers
+    func createLayerMask() -> CALayer {
+        let maskLayer = CAGradientLayer()
+        maskLayer.anchorPoint = .zero
+        maskLayer.startPoint = CGPoint(x: 0, y: 1)
+        maskLayer.endPoint = CGPoint(x: 1, y: 1)
+        
+        let lightColor = UIColor(white: 1, alpha: 0.5)
+        let darkColor = UIColor(white: 1, alpha: 1)
+        
+        maskLayer.colors = [lightColor.cgColor, darkColor.cgColor, lightColor.cgColor]
+        // If we are viewing in InterfaceBuilder we want to shift the gradient to the center so that it can be seen at design time otherwise set it to the far left.
+        if (showLoadingViewsInStoryboard && isDisplayingInStoryboard) || isSnapshotTesting {
```

Comment:
Similar to above comment about `isDisplayingInStoryboard` and `isSnapshotTesting` - maybe `showLoadingViewsInStoryboard` could be combined into that as well?

Reply:
Changed to use new isGradientCentered private var.